dlp/deid.py

```diff
@@ -0,0 +1,548 @@
+    print(response.item.value)
+# [END deidentify_with_fpe]
+
+
+# [START reidentify_with_fpe]
+def reidentify_with_fpe(parent, string, alphabet=None,
+                        surrogate_type=None, key_name=None, wrapped_key=None):
+    """Uses the Data Loss Prevention API to reidentify sensitive data in a
+    string that was encrypted by Format Preserving Encryption (FPE).
+    Args:
+        item: The string to deidentify (will be treated as text).
+        alphabet: The set of characters to replace sensitive ones with. For
+            more information, see https://cloud.google.com/dlp/docs/reference/
+            rest/v2beta2/organizations.deidentifyTemplates#ffxcommonnativealphabet
+        surrogate_type: The name of the surrogate custom info type to used
+            during the encryption process.
+        key_name: The name of the Cloud KMS key used to encrypt ('wrap') the
+            AES-256 key. Example:
+            keyName = 'projects/YOUR_GCLOUD_PROJECT/locations/YOUR_LOCATION/
+            keyRings/YOUR_KEYRING_NAME/cryptoKeys/YOUR_KEY_NAME'
+        wrapped_key: The encrypted ('wrapped') AES-256 key to use. This key
+            should be encrypted using the Cloud KMS key specified by key_name.
+    Returns:
+        None; the response from the API is printed to the terminal.
+    """
+    # Import the client library
+    import google.cloud.dlp
+
+    # Instantiate a client
+    google.cloud.dlp.DlpServiceClient.SERVICE_ADDRESS = 'autopush-dlp.sandbox.googleapis.com'
+    dlp = google.cloud.dlp.DlpServiceClient()
+
+    # Add parent
+    parent = dlp.project_path(parent)
+
+    # Wrapped key can not be base64 encoded
+    import base64
+    wrapped_key = base64.b64decode(wrapped_key)
+
+    # Construct Deidentify Config
+    reidentify_config = {
+        'info_type_transformations': {
+            'transformations': [
+                {
+                    'primitive_transformation': {
+                        'crypto_replace_ffx_fpe_config': {
+                            'crypto_key': {
+                                'kms_wrapped': {
+                                    'wrapped_key': wrapped_key,
+                                    'crypto_key_name': key_name
+                                }
+                            },
+                            'common_alphabet': alphabet,
+                            'surrogate_info_type': {
+                                'name': surrogate_type
+                            }
+                        }
+                    }
+                }
+            ]
+        }
+    }
+
+    inspect_config = {
+        'custom_info_types': [
+            {
+                'info_type': {
+                    'name': surrogate_type
+                },
+                'surrogate_type': {
+                }
+            }
+        ]
+    }
+
+    # Convert string to item
+    item = {'value': string}
+
+    # Call the API
+    response = dlp.reidentify_content(parent,
```

Comment:
please avoid hanging indents throughout, prefer to line break at the opening `(` and then indent the continuation by one level.

Reply:
done